### PR TITLE
BUG: addresses #14855 by fixing color kwarg conflict

### LINKED ISSUE #14855: scatter_matrix `color` vs `c`
I reported the issue below to matplotlib [here](https://github.com/matplotlib/matplotlib/issues/7603#issuecomment-266250371), but they have requested to report it here as it appears (to the matplotlib folks) related to pandas.

The following code gives error, but works replacing `color=` with `c=`

I suspect is either a bug, or some piece of documentation missing?
(matplotlib folks think it is a pandas / scatter_matrix bug - see below)

I'm using:
matplotlib 1.5.3
pandas 0.19.1
python 3.5

This is a sample code:
```
import pandas as pd
import numpy as np
from sklearn.datasets import load_iris
%matplotlib inline

iris = load_iris()
colors = list()
palette = {0: "red", 1: "green", 2: "blue"}

for c in np.nditer(iris.target): colors.append(palette[int(c)])
    # using the palette dictionary, we convert
    # each numeric class into a color string
dataframe = pd.DataFrame(iris.data,
columns=iris.feature_names)
scatterplot = pd.scatter_matrix(dataframe, alpha=0.3,
figsize=(10, 10), diagonal='hist', color=colors, marker='o', grid=True)
```
This is the error trace:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-f3d30057e124> in <module>()
     14 columns=iris.feature_names)
     15 scatterplot = pd.scatter_matrix(dataframe, alpha=0.3,
---> 16 figsize=(10, 10), diagonal='hist', color=colors, marker='o', grid=True)

/Users/e/anaconda/lib/python3.5/site-packages/pandas/tools/plotting.py in scatter_matrix(frame, alpha, figsize, ax, grid, diagonal, marker, density_kwds, hist_kwds, range_padding, **kwds)
    393 
    394                 ax.scatter(df[b][common], df[a][common],
--> 395                            marker=marker, alpha=alpha, **kwds)
    396 
    397                 ax.set_xlim(boundaries_list[j])

/Users/e/anaconda/lib/python3.5/site-packages/matplotlib/__init__.py in inner(ax, *args, **kwargs)
   1817                     warnings.warn(msg % (label_namer, func.__name__),
   1818                                   RuntimeWarning, stacklevel=2)
-> 1819             return func(ax, *args, **kwargs)
   1820         pre_doc = inner.__doc__
   1821         if pre_doc is None:

/Users/e/anaconda/lib/python3.5/site-packages/matplotlib/axes/_axes.py in scatter(self, x, y, s, c, marker, cmap, norm, vmin, vmax, alpha, linewidths, verts, edgecolors, **kwargs)
   3787                 facecolors = co
   3788             if c is not None:
-> 3789                 raise ValueError("Supply a 'c' kwarg or a 'color' kwarg"
   3790                                  " but not both; they differ but"
   3791                                  " their functionalities overlap.")

ValueError: Supply a 'c' kwarg or a 'color' kwarg but not both; they differ but their functionalities overlap.
```
According to @goyodiaz:

> It looks like a pandas issue. pandas.scatter_matrix always pass c to scatter but the caller can optionally pass more keywords arguments, including color. If the caller pass only color then scatter_matrix will pass both c and color and matplotlib will raise with a message that is confusing to the user because they did not pass c. Probably scatter_matrix should either raise with a clearer message or drop the c keyword argument when the user pass color

.

### PR DESCRIPTION
**Pull request by alexandercbooth**
- [x] closes #14855 
 - [x] tests passed
 - [x] passes ``git diff upstream/master | flake8 --diff``

### CODE REVIEW

File: pandas/tools/plotting.py [hunk 1]
```diff
@@ -359,7 +359,10 @@ def scatter_matrix(frame, alpha=0.5, figsize=None, ax=None, grid=False,
     density_kwds = density_kwds or {}
 
     # workaround because `c='b'` is hardcoded in matplotlibs scatter method
-    kwds.setdefault('c', plt.rcParams['patch.facecolor'])
+    if 'color' in kwds:
+        pass
+    else:
```
Comment (jreback) [inline pandas/tools/plotting.py]: ``kwds.setdefault('c', kwdgs.pop('color', plt.rcParams['patch.facecolor']))`` is a big more idiomatic

**Comment by jreback** [comment]
pls add a whatsnew note (for 0.19.2 is fine)

File: pandas/tools/plotting.py (hunk 1 above)
Comment (jreback) [inline pandas/tools/plotting.py]: nvm, I see that these kwards are *slightly* different. Can you update / add to the comment (and reference the issue).

**Comment by jreback** [comment]
pls add a test

File: pandas/tools/plotting.py [hunk 2]
```diff
@@ -359,7 +359,10 @@ def scatter_matrix(frame, alpha=0.5, figsize=None, ax=None, grid=False,
     density_kwds = density_kwds or {}
 
     # workaround because `c='b'` is hardcoded in matplotlibs scatter method
-    kwds.setdefault('c', plt.rcParams['patch.facecolor'])
+    if 'color' in kwds:
```
Comment (sinhrks) [inline pandas/tools/plotting.py]: shouldn't overwrite if any of ‘c’, ‘color’ and ‘facecolor’.

**Review by sinhrks — CHANGES_REQUESTED** [review]

File: pandas/tools/plotting.py (hunk 2 above)
Comment (alexandercbooth) [inline pandas/tools/plotting.py]: I'm sorry, I don't quite understand. Could you explain a little bit more?

**Comment by alexandercbooth** [comment]
Sorry for the slow response, long week at work. Thank you both for the feedback, let me know if there's something else I can change.

File: pandas/tools/plotting.py (hunk 2 above)
Comment (sinhrks) [inline pandas/tools/plotting.py]: I meant, ``plt.scatter(..., facecolor='red')`` and ``plt.scatter(..., c='red')`` both draws points in red. So ``scatter_matrix(..., facecolor='red')`` and ``scatter_matrix(..., c='red')`` should be the same.

Comment (alexandercbooth) [inline pandas/tools/plotting.py]: Ahh gotcha. Thanks!
So one of the problems turned out that difference between `c` and `color` was an edge border on the markers. We can take care of that by setting the default to `'none'`. After making the changes here is the current behavior:
image
image
image
All are now the same.

**Comment by alexandercbooth** [comment]
I also notice that this exact same bug is present in scatter_plot. It can be fixed in exactly the same way as I have done above:

image
image
image

I would be happy to fix it as well. I could just tack it on here or open another PR, whichever is best.

File: pandas/tools/plotting.py [hunk 3]
```diff
@@ -359,7 +360,16 @@ def scatter_matrix(frame, alpha=0.5, figsize=None, ax=None, grid=False,
...
+    # only set default for `c` if `color` kwarg not supplied
+    # edgecolors default required otherwise `color` has a border
+    # GH 14855
+    kwds.setdefault('edgecolors', 'none')
+    if 'color' in kwds:
+        pass
+    if 'facecolor' in kwds:
+        kwds.setdefault('facecolor', plt.rcParams['patch.facecolor'])
+    if 'c' in kwds:
+        kwds.setdefault('c', plt.rcParams['patch.facecolor'])
```
Comment (tacaswell) [inline pandas/tools/plotting.py]: I do not understand this, if `'c'` is in kwds, then `setdefault` is a no-op.

You probably also want to version guard this, mpl 2.0 scatter respects the color cycle
http://matplotlib.org/2.0.0rc1/users/dflt_style_changes.html#scatter

**Review by tacaswell — CHANGES_REQUESTED** [review]

File: pandas/tools/plotting.py (hunk 3 above)
Comment (alexandercbooth) [inline pandas/tools/plotting.py]: Good thinking, @tacaswell! I investigated it a bit further and it turns out we can resolve this issue and the same issue for scatter_plot by removing the old bug fix, removing the new changes and adding in the edgecolor='none' as default. With this, we still get the identical behavior with `c`, `color` or `facecolor` set and no error.

**Comment by tacaswell** [comment]
The change to the default value of `c` went in via https://github.com/matplotlib/matplotlib/pull/4079 which is in mpl >=1.5.0.  pandas will need to do some version checking to correctly support older versions of mpl.  I thought back to 1.3.1 is currently supported?

**Comment by jreback** [comment]
status of this?

File: pandas/tests/plotting/test_misc.py [hunk 4]
```diff
@@ -77,6 +77,12 @@ def scat(**kwds):
             _check_plot_works(scat, diagonal='hist')
         with tm.assert_produces_warning(UserWarning):
             _check_plot_works(scat, range_padding=.1)
+        with tm.assert_produces_warning(UserWarning):
+            _check_plot_works(scat, color='rgb')
```
Comment (jorisvandenbossche) [inline pandas/tests/plotting/test_misc.py]: Is there a reason you pass here 'rgb' and not just 'r'? As passing multiple colors seems a bit odd for a single scatter plot (I think now the points just get the [r, g, b, r, g, b, r, g, b, ...] colors sequentially?)

**Review by jorisvandenbossche — COMMENTED** [review]
This seems to work (I don't fully understand how passing facecolor solves the color cs c conflict). 
The only thing is that this slightly changes the plot appearance. As previously `c` only changed the facecolor, while it now also seems to change the edgecolor.

**Review by sinhrks — APPROVED** [review]

**Comment by jreback** [comment]
status on this? @sinhrks @jorisvandenbossche @TomAugspurger

**Comment by alexandercbooth** [comment]
Thanks Jeff and everyone else for your patience on this; it's been a really hectic month at work. I appreciate all the feedback from everyone. I will try to address all concerns I can tomorrow and report back.
Thanks!

File: pandas/tests/plotting/test_misc.py (hunk 4 above)
Comment (alexandercbooth) [inline pandas/tests/plotting/test_misc.py]: Purely for aesthetic purposes. I have trouble seeing some colors and wanted to visually confirm these changes rendered the same objects.

**Comment by alexandercbooth** [comment]
I squashed and rebased since the PR was stale, but it's my first time doing that so I apologize if I made a mistake.

**Comment by jreback** [comment]
@alexandercbooth can you rebase

@TomAugspurger ok on this?

**Comment by jreback** [comment]
lgtm. @TomAugspurger ok? (don't merge yet)

**Comment by jreback** [comment]
@TomAugspurger ?

**Comment by TomAugspurger** [comment]
Sorry for the delay, +1 on this.

@jreback, when you merge PRs that need a rebase (conflicts in whatsnew), you use the `merge-pr.py` script, right? I can do it, just want to double check first. Don't want to make @alexandercbooth rebase again.

**Comment by jreback** [comment]
@TomAugspurger yes you can use the script. it will stop when it can't auto rebase (it will squash), you then fix the conflicts, ``git add``, then continue the script. Then I stop again (its now rebased), just to make sure the commits message and everything is ok. Then have it push upstream.

**Comment by TomAugspurger** [comment]
@alexandercbooth thanks!

**Comment by alexandercbooth** [comment]
My pleasure!

**Pull request closed without merging**